homeassistant/components/mqtt/__init__.py

```diff
@@ -673,8 +718,9 @@ def async_disconnect(self):
         """
         def stop():
             """Stop the MQTT client."""
+            if self._task_reconnect:
+                self._task_reconnect.cancel()
```

Comment:
Is it necessary to cancel the reconnect task (it was not done in threaded mode)?

Reply:
I think we should cancel it like this. Otherwise it could still be running after we stop home assistant which would generate an error when we close the loop.

Reply:
OK. I updated the reconnect loop to catch asyncio.CancelledError